File: frontend-web/src/services/authService.ts
import type { Usuario } from '../types'
import { apiClient } from '../api/client'

export interface LoginResponse {
  token: string
  usuario: Usuario
}

export function login(email: string, password: string): Promise<LoginResponse> {
  return apiClient.post<LoginResponse>('/auth/login', { email, password })
}

File: appmovil/lib/services/auth_service.dart
import '../models/usuario.dart';
import 'api_client.dart';

class AuthResult {
  final String token;
  final Usuario usuario;

  const AuthResult({required this.token, required this.usuario});
}

class AuthService {
  final ApiClient _api;

  AuthService(this._api);

  Future<AuthResult> login(String email, String password) async {
    final data = await _api.post(
      '/api/auth/login',
      {'email': email, 'password': password},
      isAuthCall: true,
    );
    final json = data as Map<String, dynamic>;
    return AuthResult(
      token: json['token'] as String,
      usuario: Usuario.fromJson(json['usuario'] as Map<String, dynamic>),
    );
  }
}
